Extension/src/LanguageServer/Providers/semanticTokensProvider.ts

```diff
@@ -18,7 +18,7 @@ export class SemanticTokensProvider implements vscode.DocumentSemanticTokensProv
 
     public async provideDocumentSemanticTokens(document: vscode.TextDocument, token: vscode.CancellationToken): Promise<vscode.SemanticTokens> {
         return new Promise<vscode.SemanticTokens>((resolve, reject) => {
```

Comment:
Another.  I think this Promise is unnecessary.  This function is already async.

Reply:
now `return new Promise` are removed in other places wherever the function is async.